providers/os/resources/fstab.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+package resources
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+
+	"go.mondoo.com/cnquery/v11/llx"
+	"go.mondoo.com/cnquery/v11/providers/os/connection/shared"
+)
+
+func (f *mqlFstab) entries() ([]any, error) {
+	conn := f.MqlRuntime.Connection.(shared.Connection)
+
+	fs := conn.FileSystem()
+	if fs == nil {
+		return nil, errors.New("filesystem not available")
+	}
+
+	fstabFile, err := fs.Open("/etc/fstab")
```

Comment:
can we make this a param of the actual MQL resource? that will let us read fstabs on snapshots too

Reply:
It does read fstab on snapshots already(example in PR description), since mounted dir is handled by FileSystem connecion.
I will add the init() for the edge case where /etc is mounted separately, it's probably going to be an issue eitherway